core/lib/dr_events.h

```diff
@@ -641,6 +641,12 @@ typedef struct _dr_fault_fragment_info_t {
      * depending on the type of cache consistency being used by DR.
      */
     bool app_code_consistent;
+    /**
+     * The recreated ilist for this fragment. This also contains metadata
+     * added by various clients, which may be helpful in restoring state
+     * on fault. Set to NULL if not available.
+     */
+    instrlist_t *ilist;
```

Comment:
Hmm we don't have a size field here: so I guess a new client with an old DR will have trouble, but that's not a major concern: mostly it's an old client with a new DR which would be fine.

Reply:
Old client and new DR should do okay. The new field is the last member of `dr_fault_fragment_info_t` which is also the last member of `dr_restore_state_info_t`, so the old client should just ignore the new ilist pointer.